Modules/mmapmodule.c

```diff
@@ -607,18 +637,34 @@ mmap_gfind(mmap_object *self,
     }
 }
 
+/*[clinic input]
+@critical_section
+mmap.mmap.find
+
+  *args: tuple
+
+[clinic start generated code]*/
+
 static PyObject *
-mmap_find_method(PyObject *op, PyObject *args)
+mmap_mmap_find_impl(mmap_object *self, PyObject *args)
+/*[clinic end generated code: output=6e0dfd51873d7263 input=dc56119de60c458f]*/
 {
-    mmap_object *self = mmap_object_CAST(op);
-    return mmap_gfind(self, args, 0);
+    return mmap_gfind_lock_held(self, args, 0);
 }
 
+/*[clinic input]
+@critical_section
+mmap.mmap.rfind
+
+  *args: tuple
```

Comment:
Same, refactoring `mmap_gfind_lock_held` to pass the buffer to find directly instead of parsing would have a much better signature.